core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -419,4 +429,45 @@ public void testCanReadOldCompressedManifestFiles() throws Exception {
     List<FileScanTask> tasks = Lists.newArrayList(reloaded.newScan().planFiles());
     Assert.assertEquals("Should scan 1 files", 1, tasks.size());
   }
+
+  @Test
+  public void testConcurrentAppend() throws Exception {
+    assertTrue("Should create v1 metadata",
+            version(1).exists() && version(1).isFile());
+    File dir = temp.newFolder();
+    dir.delete();
+    int threadsCount = 5;
+    int numberOfCommitedFilesPerThread = 10;
+    Table tableWithHighRetries = TABLES.create(SCHEMA, SPEC,
+        ImmutableMap.of(COMMIT_NUM_RETRIES, String.valueOf(threadsCount)), dir.toURI().toString());
+    String fileName = UUID.randomUUID().toString();
+    DataFile file = DataFiles.builder(tableWithHighRetries.spec())
+        .withPath(FileFormat.PARQUET.addExtension(fileName))
+        .withRecordCount(2)
+        .withFileSizeInBytes(0)
+        .build();
+    ExecutorService executorService = Executors.newFixedThreadPool(threadsCount);
+    AtomicInteger barrier = new AtomicInteger(0);
+    Tasks
+        .range(threadsCount)
+        .stopOnFailure()
+        .throwFailureWhenFinished()
+        .executeWith(executorService)
+        .run(index -> {
+          for (int numCommittedFiles = 0; numCommittedFiles < numberOfCommitedFilesPerThread; numCommittedFiles++) {
+            while (barrier.get() < numCommittedFiles * threadsCount) {
+              try {
+                Thread.sleep(10);
+              } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            tableWithHighRetries.newFastAppend().appendFile(file).commit();
+            barrier.incrementAndGet();
+          }
+        });
+    tableWithHighRetries.refresh();
```

Comment:
Can you add newlines to match the test that this was copied from? There are no newlines and we typically like to break up the code more than this.